src/main/java/com/bobocode/petros/container/ApplicationAnnotationContainer.java

```diff
@@ -66,9 +70,13 @@ private DependencyDefinition keyByDependencyDefinitionName(String name) {
 
     @SuppressWarnings("unchecked")
     private <T> T getDependencyFromMap(Class<T> clazz) {
-        return (T) dependencyMap.values().stream()
-                .filter(obj -> obj.getClass().equals(clazz))
-                .findFirst()
-                .orElseThrow();
+        if(!isNonUniqueDependency(clazz)) {
+            return (T) dependencyMap.values().stream()
+                    .filter(obj -> obj.getClass().equals(clazz))
+                    .findFirst()
+                    .orElseThrow();
+        } else {
```

Comment:
If and else is +1 level of abstraction, usually if possible trying to get rid of it.
For example in this case:

```
private <T> T getDependencyFromMap(Class<T> clazz) {
  if (!isNonUniqueDependency(clazz)) {
     throw new NoUniqueDependencyException(clazz.getName());
  }
  
  return (T) dependencyMap.values().stream()
                    .filter(obj -> obj.getClass().equals(clazz))
                    .findFirst()
                    .orElseThrow();
}
```
logically it should be the same, but in term of reading and maintaining it's more easier.

Reply:
Yeah, nice catch)